File: metaurl/handler.js
"use strict";

//TODO: Extract body?

const metascraper = require("metascraper")([
  require("metascraper-author")(),
  require("metascraper-date")(),
  require("metascraper-description")(),
  require("metascraper-image")(),
  require("metascraper-logo")(),
  require("metascraper-clearbit")(),
  require("metascraper-publisher")(),
  require("metascraper-title")(),
  require("metascraper-url")(),
]);

const got = require("got");

module.exports = async (event, context) => {
  const targetUrl = event.query.url;

  try {
    const { body: html, url } = await got(targetUrl, { timeout: 5000 });
    const metadata = await metascraper({ html, url });
    return context
      .status(200)
      .headers({ "Content-Type": "application/json" })
      .succeed(JSON.stringify(metadata));
  } catch (e) {
    return context
      .status(422)
      .headers({ "Content-Type": "application/json" })
      .fail(JSON.stringify({ error: "Unprocessable entity" }));
  }
};

File: readability/handler.py
from newspaper import Article


def handle(event, context):
    url = event.query['url']

    article = Article(url)
    article.download()
    article.parse()

    result = {
        "text": article.text,
        "description": article.meta_data["og"].get("description"),
        "meta_data": article.meta_data,
        "title": article.title,
        "image": article.top_image,
        "url": url
    }

    return {
        "statusCode": 200,
        "body": result
    }
